model-engine/model_engine_server/infra/gateways/gcs_filesystem_gateway.py

```diff
@@ -0,0 +1,46 @@
+import os
+import re
+from typing import IO, Optional, Dict
+
+import smart_open
+from google.cloud import storage
+from model_engine_server.infra.gateways.filesystem_gateway import FilesystemGateway
+
+
+class GCSFilesystemGateway(FilesystemGateway):
+    """
+    Concrete implementation for interacting with Google Cloud Storage.
+    """
+
+    def get_storage_client(self, kwargs: Optional[Dict]) -> storage.Client:
+        """
+        Retrieve or create a Google Cloud Storage client. Could optionally
+        utilize environment variables or passed-in credentials.
+        """
+        project = kwargs.get("gcp_project", os.getenv("GCP_PROJECT"))
```

Comment:
where does this env var get set? it seems analogous to `AWS_PROFILE` but those changes would need to be baked into any relevant k8s yamls most likely